### PR TITLE
 [ContainerApp] az containerapp replica count: app/revision scope 

### PR DESCRIPTION
**Pull request by Juancpani**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ContainerApp

**Comment by Juancpani** [comment]
Can I get reviews on this PR? @Greedygre, @Juliehzl

File: src/containerapp/azext_containerapp/commands.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def load_command_table(self, _):
     with self.command_group('containerapp replica') as g:
         g.custom_show_command('show', 'get_replica')  # TODO implement the table transformer
         g.custom_command('list', 'list_replicas')
+        g.custom_command('count', 'count_replicas', is_preview=True)
```
Comment (pagariyaalok) [inline src/containerapp/azext_containerapp/commands.py:35]: >is_preview=True [](http://example.com/codeflow?start=52&length=15)

why is this preview ?

File: src/containerapp/azext_containerapp/custom.py [hunk 2]
```diff
@@ -3347,24 +3347,53 @@ def remove_dapr_component(cmd, resource_group_name, dapr_component_name, environ
...
+def count_replicas(cmd, resource_group_name, name, revision=None):
+
+    try:
+        app = ContainerAppClient.show(cmd, resource_group_name, name)
+        if not revision:
+            revision = safe_get(app, "properties", "latestRevisionName")
+            if not revision:
+                raise ValidationError("No revision found for containerapp.")
+    except Exception as e:
+        handle_raw_exception(e)
```
Comment (pagariyaalok) [inline src/containerapp/azext_containerapp/custom.py:3372]: Do we need separate try catch for this ?

**Review by pagariyaalok — APPROVED** [review]
:shipit:

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/commands.py:35]: Hi @pagariyaalok 

> why is this preview ?

We have containerapp in core https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/containerapp, we only put GA commands in core without `is_preview=True`.
About the iteration process for new commands:
Firstly, add new commands or new features in containerapp extension with tag `is_preview=True`.
After released in extension and being mature enough, you can move it to the core, at that time you can remove the `is_preview=True` in extension.

File: src/containerapp/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,7 @@
 Release History
 ===============
 upcoming
+* 'az containerapp replica count', returns the replica count of a container app
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:6]:
```suggestion
++++++
* 'az containerapp replica count', returns the replica count of a container app
```

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (Juancpani) [reply src/containerapp/azext_containerapp/commands.py:35]: Thank you, Xinyu!

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you please help to review this PR? Thanks.

**Review by yanzhudd — APPROVED** [review]

File: projects
Comment (zhoxing-ms) [inline projects:1]: Just out of curiosity, I would like to confirm why the `projects` folder was added at that time?

Comment (Greedygre) [reply projects:1]: Hi @Juancpani 
Is this `projects` folder added by mistake?

Comment (Juancpani) [reply projects:1]: Yes, the projects folder addition was a mistake. We should remove it.

**Pull request merged**